src/System.Management.Automation/engine/CommandCompletion/ExtensibleCompletion.cs

```diff
@@ -158,4 +158,56 @@ protected override void EndProcessing()
             }
         }
     }
+
+    /// <summary>
+    /// This attribute is used to specify an argument completions for a parameter of a cmdlet or function
+    /// based on string array.
+    /// <example>
+    ///     [Parameter()]
+    ///     [ArgumentCompletions("Option1","Option2","Option3")]
+    ///     public string Noun { get; set; }
+    /// </example>
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
+    public class ArgumentCompletionsAttribute : Attribute
+    {
+        private string[] _completions;
+
+        /// <summary>
+        /// Initializes a new instance of the ArgumentCompleterAttribute class
+        /// </summary>
+        /// <param name="completions">list of complete values</param>
+        /// <exception cref="ArgumentNullException">for null arguments</exception>
+        /// <exception cref="ArgumentOutOfRangeException">for invalid arguments</exception>
+        public ArgumentCompletionsAttribute(params string[] completions)
+        {
+            if (completions == null)
+            {
+                throw PSTraceSource.NewArgumentNullException("completions");
+            }
+
+            if (completions.Length == 0)
+            {
+                throw PSTraceSource.NewArgumentOutOfRangeException("completions", completions);
+            }
+
+            _completions = completions;
+        }
+
+        /// <summary>
+        /// The function returns completions for arguments.
+        /// </summary>
+        public IEnumerable<CompletionResult> CompleteArgument(string commandName, string parameterName, string wordToComplete, CommandAst commandAst, IDictionary fakeBoundParameters)
+        {
+            var wordToCompletePattern = WildcardPattern.Get(string.IsNullOrWhiteSpace(wordToComplete) ? "*" : wordToComplete + "*", WildcardOptions.IgnoreCase);
+
+            foreach (var str in _completions)
+            {
+                if (wordToCompletePattern.IsMatch(str))
+                {
+                    yield return new CompletionResult(str, str, CompletionResultType.Text, str);
```

Comment:
Should this be `CompletionResultType.ParameterValue`? I think that's the only place the attribute will have any effect.

Reply:
I agree. 
Fixed.